File: crates/qgis-sys/src/core/application/info.rs
#[cxx::bridge(namespace = "qgis_shim::core")]
pub mod ffi {
    #[derive(Debug)]
    pub struct AppInfo {
        pub version: String,
        pub qt_version: String,
        pub platform: String,
    }

    unsafe extern "C++" {
        include!("qgis-sys/include/core/application_info.h");
        fn application_info() -> AppInfo;
    }
}

File: crates/qgis-sys/src/core/application/info.cpp
#include <qgsconfig.h>

#include <QSysInfo>
#include <QtGlobal>

#include "qgis-sys/include/core/application_info.h"
#include "qgis-sys/include/core/convert.h"

namespace qgis_shim::core {

AppInfo application_info() noexcept {
    AppInfo out{};
    try {
        out.version = rust::String(_QGIS_VERSION);
        out.qt_version = rust::String(qVersion());
        out.platform = to_rust(QSysInfo::prettyProductName());
        return out;
    } catch (...) {
        return AppInfo{};
    }
}

}  // namespace qgis_shim::core
